File: BAEKJOON/Python/Two Pointer/(1806)부분합.py
if __name__=="__main__":
    N, S = map(int, input().split(' '))

    # index error 방지용 [0]
    num_list = list(map(int, input().split(' '))) + [0]

    start_pointer = 0
    end_pointer = 0
    partial_sum = num_list[start_pointer]

    answer = 2 << 31

    while end_pointer < N:
        if partial_sum < S:
            end_pointer += 1
            partial_sum += num_list[end_pointer]
        else:
            answer = min(answer, end_pointer - start_pointer + 1)
            partial_sum -= num_list[start_pointer]
            start_pointer += 1

    if answer == 2 << 31:
        print(0)
    else:
        print(answer)

File: BAEKJOON/C++/Two Pointer/(1806)부분합.cpp
#include <iostream>
#include <algorithm>

using namespace std;

#define MAXNUM 100001 

int arr[100000];
int answer = MAXNUM;

bool compare(int a, int b)
{
	return a < b;
}

int main(int argc, char* argv[])
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int N, S, num;

	// 입력
	cin >> N >> S;
	for (int i = 0; i < N; i++)
	{
		cin >> num;
		arr[i] = num;
	}

	// two pointer
	int start_pointer = 0;
	int end_pointer = 0;
	int partial_sum = arr[start_pointer];

	// answer
	while (end_pointer < N)
	{	
		// 부분 합 구하기
		if (partial_sum < S)
		{
			end_pointer += 1;
			partial_sum += arr[end_pointer];
		}
		else
		{
			answer = min(answer, end_pointer - start_pointer + 1);
			partial_sum -= arr[start_pointer];
			start_pointer += 1;
		}
	}

	if (answer == MAXNUM) cout << 0;
	else cout << answer;

	return 0;
}